samples/apps/copilot-chat-app/webapi/CopilotChat/Skills/ChatSkills/CopilotChatPlanner.cs

```diff
@@ -43,21 +47,84 @@ public CopilotChatPlanner(IKernel plannerKernel, PlannerOptions? plannerOptions)
     /// Create a plan for a goal.
     /// </summary>
     /// <param name="goal">The goal to create a plan for.</param>
+    /// <param name="availableFunctionsOnly">Optional override to allow developers to keep functions regardless of whether they're available in the planner's context or not.</param>
     /// <returns>The plan.</returns>
-    public Task<Plan> CreatePlanAsync(string goal)
+    public async Task<Plan> CreatePlanAsync(string goal, bool availableFunctionsOnly = true)
     {
         FunctionsView plannerFunctionsView = this.Kernel.Skills.GetFunctionsView(true, true);
         if (plannerFunctionsView.NativeFunctions.IsEmpty && plannerFunctionsView.SemanticFunctions.IsEmpty)
         {
             // No functions are available - return an empty plan.
-            return Task.FromResult(new Plan(goal));
+            return new Plan(goal);
         }
 
-        if (this._plannerOptions?.Type == PlanType.Sequential)
+        Plan plan = this._plannerOptions?.Type == PlanType.Sequential
+                ? await new SequentialPlanner(this.Kernel, new SequentialPlannerConfig { RelevancyThreshold = 0.75 }).CreatePlanAsync(goal)
+                : await new ActionPlanner(this.Kernel).CreatePlanAsync(goal);
+
+        return availableFunctionsOnly ? this.SanitizePlan(plan, plannerFunctionsView) : plan;
+    }
+
+    #region Private
+
+    /// <summary>
+    /// Scrubs plan of functions not available in Planner's kernel.
+    /// </summary>
+    private Plan SanitizePlan(Plan plan, FunctionsView availableFunctions)
```

Comment:
I worry about the precedent we set by doing this. We really shouldn't have to. What do you think if we update SequentialPlanner to throw exception on plan creation if it is unable to create a plan with the available functions?

Reply:
I think the better solution is to fix the hallucinations, since the main problem roots from the planner pulling in functions listed as examples in the prompt. It more-than-intermittently creates SequentialPlans with these example functions.

Reply:
Generally, there's value in checking for input / parameter dependencies before returning the plan to the user, whether we mark it to prompt for user intervention or fail the plan completely.

Reply:
> more-than-intermittently creates

With what model? Gpt-4?

Reply:
Yeah, it happens when using gpt-4 especially if the ask explicitly signals to a function that's listed as an example (but not enabled yet)

i.e., have not enabled the Graph connector so the two empty step plans are likely 
1. _GLOBAL_FUNCTIONS_.GetEmailAddress
2. _GLOBAL_FUNCTIONS_.SendEmail

![image](https://github.com/microsoft/semantic-kernel/assets/125500434/6a302165-2a17-4310-9c5c-09d949324a42)

Also observable with gpt-3.5-turbo, especially if the relevancy is less than .50

Reply:
Resolving conversation since unavailable functions issue has been fixed here: https://github.com/microsoft/semantic-kernel/pull/1727/files